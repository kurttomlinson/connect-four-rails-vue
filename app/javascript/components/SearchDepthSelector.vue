<template>
  <div class="centered">
    <label for="search-depth-selector">Search depth:</label>
    <select
      id="search-depth-selector"
      @change="setSearchDepth">
      <template v-for="depth in [1,2,3,4,5]">
        <template v-if="depth == searchDepth">
          <option :key="depth" selected>{{ depth }}</option>
        </template>
        <template v-else>
          <option :key="depth">{{ depth }}</option>
        </template>
      </template>
    </select>
  </div>
</template>

<script>
export default {
  props: {
    searchDepth: {
      type: Number,
      required: true,
      validator: function (value) {
        return (value >= 1) || (value <= 5)
      }
    }
  },
  methods: {
    setSearchDepth (e) {
      this.$emit('setSearchDepth', parseInt(e.target.value))
    }
  }
}
</script>

<style scoped>
</style>
