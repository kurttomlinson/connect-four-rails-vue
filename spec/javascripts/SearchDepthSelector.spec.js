import { shallow } from 'vue-test-utils'
import Vue from 'vue'
import SearchDepthSelector from 'components/SearchDepthSelector'

describe('SearchDepthSelector.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(SearchDepthSelector, {
      propsData: {
        searchDepth: 3
      }
    })
  })

  it('emits a "setSearchDepth" event when an option is picked', () => {
    const stub = jest.fn()
    wrapper.vm.$on('setSearchDepth', stub)
    for (let i = 0; i < 5; i++) {
      wrapper.findAll('option').at(i).trigger('change')
      expect(stub).toBeCalledWith(i+1)
    }
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
