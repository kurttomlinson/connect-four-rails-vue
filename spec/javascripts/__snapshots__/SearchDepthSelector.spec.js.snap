// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDepthSelector.vue has the expected html structure 1`] = `
<div
  class="centered"
>
  <label
    for="search-depth-selector"
  >
    Search depth:
  </label>
   
  <select
    id="search-depth-selector"
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option
      selected="selected"
    >
      3
    </option>
    <option>
      4
    </option>
    <option>
      5
    </option>
  </select>
</div>
`;
